c_stubs/xenctrlext_stubs.c

```diff
@@ -460,6 +460,130 @@ CAMLprim value stub_xenctrlext_get_msr_arch_caps(value xch) {
 	CAMLreturn(caml_copy_int64(val));
 }
 
+#define Val_none (Val_int(0))
+static unsigned int ocaml_list_to_c_bitmap(value l)
+             /* ! */
+             /*
+	      * All calls to this function must be in a form suitable
+	      * for xenctrl_abi_check.  The parsing there is ad-hoc.
+	      */
+{
+	unsigned int val = 0;
+
+	for ( ; l != Val_none; l = Field(l, 1) )
+		val |= 1u << Int_val(Field(l, 0));
+
+	return val;
+}
+static void domain_handle_of_uuid_string(xen_domain_handle_t h,
+					 const char *uuid)
+{
+# define SCNx8       "hhx"
+#define X "%02"SCNx8
+#define UUID_FMT (X X X X "-" X X "-" X X "-" X X "-" X X X X X X)
+
+	if ( sscanf(uuid, UUID_FMT, &h[0], &h[1], &h[2], &h[3], &h[4],
+		    &h[5], &h[6], &h[7], &h[8], &h[9], &h[10], &h[11],
+		    &h[12], &h[13], &h[14], &h[15]) != 16 )
+	{
+		char buf[128];
+
+		snprintf(buf, sizeof(buf),
+			 "Xc.int_array_of_uuid_string: %s", uuid);
+
+		caml_invalid_argument(buf);
+	}
+
+#undef X
+}
+
+CAMLprim value stub_xc_domain_create_domid(value xch, value config, value want_domid)
+{
+	CAMLparam2(xch, config);
```

Comment:
Why does  want_domid not need to be declared as a parameter? Add a comment.

Reply:
Well spotted, that is a bug. I have to use CAMLparam3 here.